components/support/nimbus-cli/src/cmd.rs

```diff
@@ -318,15 +332,42 @@ impl LaunchableApp {
         Ok(true)
     }
 
-    fn start_app(&self, reset_db: bool, payload: Option<&Value>, log_state: bool) -> Result<bool> {
+    fn open(&self, deeplink: Option<&String>) -> Result<bool> {
+        self.start_app(false, None, false, deeplink)
+    }
+
+    fn create_deeplink(&self, deeplink: Option<&String>) -> Result<Option<String>> {
+        if deeplink.is_none() {
+            return Ok(None);
+        }
+        let deeplink = deeplink.unwrap();
+        Ok(if deeplink.contains("://") {
+            Some(deeplink.clone())
+        } else if let Some(scheme) = match self {
+            Self::Android { scheme, .. } | Self::Ios { scheme, .. } => scheme,
+        } {
+            Some(format!("{}://{}", scheme, deeplink))
+        } else {
+            anyhow::bail!("Cannot use a deeplink without a scheme for this app")
+        })
+    }
+
+    fn start_app(
+        &self,
+        reset_db: bool,
+        payload: Option<&Value>,
+        log_state: bool,
+        deeplink: Option<&String>,
+    ) -> Result<bool> {
+        let deeplink = self.create_deeplink(deeplink)?;
         Ok(match self {
             Self::Android { .. } => self
-                .android_start(reset_db, payload, log_state)?
+                .android_start(reset_db, payload, log_state, deeplink.as_ref())?
```

Comment:
What happens with `deeplink.as_ref()` if `deeplink` doesn't exist?

Reply:
`deeplink` as a reference will always exist, but it may not always have a value stored in it. I've kinda mentally mapped `Option` to the concept of an enum where the only values are `Some(Inner)` and `None`. Not sure if you'd find that line of thought helpful though.

If you see places where we use `deeplink.unwrap()`, `deeplink?` or `let Some(x) = deeplink`, those are all ways we unwrap an `Option` into its inner value/type.